File: angular/src/app/frontend/models/hotel-search-query-dto.ts


import { PagedAndSortedResultRequestDto } from '@abp/ng.core';

export class HotelSearchQueryDto extends PagedAndSortedResultRequestDto {
  address: string;
  checkIn: string;
  checkOut: string;
  adults: number;
  children: number;
  rooms: number;
  filter: string;
  sorting: string;
  skipCount: number;
  maxResultCount: number;

  constructor(initialValues: Partial<HotelSearchQueryDto> = {}) {
    super(initialValues);
  }
}

File: aspnet-core/src/HotelApp.Application.Contracts/Frontends/HotelSearchQueryDto.cs
using System;
using Volo.Abp.Application.Dtos;

namespace HotelApp.Frontends
{
    public class HotelSearchQueryDto : PagedAndSortedResultRequestDto
    {
        public string Address { get; set; }
        public DateTime CheckIn { get; set; }
        public DateTime CheckOut { get; set; }
        public short Adults { get; set; }
        public short Children { get; set; }
        public short Rooms { get; set; }
        public string Filter { get; set; }
    }
}
